helper/validate_password_test.go

```diff
@@ -0,0 +1,86 @@
+package helper
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Validate(t *testing.T) {
+
+	testCases := []struct {
+		name     string
+		arg      string
+		expError error
+	}{
+		{
+			name:     "success",
+			arg:      "1q?btAjhpztqnln",
+			expError: nil,
+		},
+		{
+			name:     "failed, password is too short",
+			arg:      "",
+			expError: ErrLength,
+		},
+		{
+			name:     "failed, password should contains at least one uppercase letter",
+			arg:      "1q?btajhpztqnln",
+			expError: ErrUpper,
+		},
+		{
+			name:     "failed, password should contains at least one lowercase letter",
+			arg:      "1Q?BTAJHPZTQNLTN",
+			expError: ErrLow,
+		},
+		{
+			name:     "failed, password should contains at least one digit",
+			arg:      "aq?btAjhpztqnln",
+			expError: ErrNum,
+		},
+		{
+			name:     "failed, password should contains at least one special symbol",
+			arg:      "aq1Btajhpztqnln",
+			expError: ErrSymbol,
+		},
+		{
+			name:     "failed, password should contains don't repeat parts",
+			arg:      "1234567812345678?Aa",
+			expError: fmt.Errorf("weak pass: Password is too systematic"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			v := NewPasswordValidator()
+
+			err := v.Validate(tc.arg)
+			if tc.expError != nil {
+				assert.Equal(t, tc.expError.Error(), err.Error())
+			} else {
+				assert.Nil(t, err)
+			}
+
+		})
+	}
+}
+
+func Test_Validate_search_in_dictionary(t *testing.T) {
```

Comment:
why we need test functionality from a package?

Reply:
Sorry, I don't catch what are you asking about.

I made `Test_Validate_search_in_dictionary ` to not overload `Test_Validate`.

The `Test_Validate` contains basic cases.
The `Test_Validate_search_in_dictionary` contains the dictionary-related logic.